tests/agent/actions/test_list_table_names_action.py

```diff
@@ -0,0 +1,78 @@
+import uuid
+from unittest import mock
+from unittest.mock import MagicMock
+
+import pytest
+from sqlalchemy.engine import Inspector
+
+from great_expectations.agent.actions import (
+    ListTableNamesAction,
+)
+from great_expectations.agent.models import ListTableNamesEvent
+from great_expectations.data_context import CloudDataContext
+from great_expectations.datasource.fluent import (
+    PandasDatasource,
+    SQLDatasource,
+)
+from great_expectations.exceptions import StoreBackendError
+
+pytestmark = pytest.mark.unit
+
+
+@pytest.fixture(scope="function")
+def context():
+    return MagicMock(autospec=CloudDataContext)
+
+
+@pytest.fixture
+def event():
+    return ListTableNamesEvent(
+        type="list_table_names_request.received",
+        datasource_name="test-datasource",
+    )
+
+
+def test_list_table_names_event_raises_for_non_sql_datasource(context, event):
+    action = ListTableNamesAction(context=context)
+    id = "096ce840-7aa8-45d1-9e64-2833948f4ae8"
+    context.get_expectation_suite.side_effect = StoreBackendError("test-message")
+    context.get_checkpoint.side_effect = StoreBackendError("test-message")
+    datasource = MagicMock(spec=PandasDatasource)
+    context.get_datasource.return_value = datasource
+
+    with pytest.raises(
+        TypeError, match=r"This operation requires a SQL Datasource but got"
+    ):
+        action.run(event=event, id=id)
+
+
+def test_run_list_table_names_action_returns_action_result(context, event):
+    action = ListTableNamesAction(context=context)
+    id = "096ce840-7aa8-45d1-9e64-2833948f4ae8"
+
+    with mock.patch(
+        "great_expectations.agent.actions.list_table_names.inspect"
+    ) as mock_inspect, mock.patch(
+        "requests.Session.patch", autospec=True
+    ) as mock_patch:
+        datasource = MagicMock(spec=SQLDatasource)
+        datasource_id = str(uuid.uuid4())
+        datasource.id = datasource_id
+        context.get_datasource.return_value = datasource
+
+        mock_resp = MagicMock()
+        mock_resp.json.return_value = ""
+        mock_resp.status_code = 204
+        mock_patch.return_value = mock_resp
```

Comment:
`non-blocking`

I know you've been trying to get this in for a while but maybe as a follow up we could pair on using `responses` for this HTTP response mocks.

https://github.com/getsentry/responses#basics
https://github.com/getsentry/responses#dynamic-responses

Reply:
Yes please! I'll put something on calendar.